playbooks/roles/mongo_3_2/templates/backup-mongo.sh.j2

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# ref https://tasks.opencraft.com/browse/SE-1669
+# Script to perform a point-in-time dump of the local mongodb database using
+# mongodump.
+# includes locking (prevent this script from running multiple times in
+# parallel), creating a snapshot of the volume used to backup to, and snitch monitoring
+
+# vars
+ARCHIVE_DIR="TODO"
+ARCHIVE_NAME="mongo-$(date --iso-8601=minutes --utc).gz"
+MONGO_ADMIN_PASSWORD="TODO"
+VOLUME="TODO" # need some way to tell the script which is the volume being used - must be a dedicated ebs
+
+
+# TODO: lock
```

Comment:
Recommended approach is to [use the script itself as a lock file](https://github.com/open-craft/ansible-playbooks/blob/f05caa987039b71bff737b7ae6fd71aefccc56bd/playbooks/roles/tarsnap/templates/backup.sh#L4-L7).

Reply:
Ah that's exactly what I was looking for - I remember seeing that earlier. Thanks!